lib/utils/source_ip_check.ts

```diff
@@ -116,25 +118,36 @@ export function checkSourceIpForHandler<P1, P2, P3, P4, P5, P6, O>(
   p4: P4,
   p5: P5,
   p6: P6
-) => Promise<O | IResponseErrorForbiddenNotAuthorized> {
+) => Promise<
+  O | IResponseErrorForbiddenNotAuthorized | IResponseErrorInternal
+> {
   return (p1: P1, p2: P2, p3: P3, p4: P4, p5: P5, p6: P6) => {
-    // extract the x-forwarded-for header and the allowed cidrs from the params
-    const x = extractor(p1, p2, p3, p4, p5, p6);
-    const maybeClientIp = x.e1;
-    const cidrs = x.e2;
+    return new Promise(resolve => {
```

Comment:
why you need this? there's no callback involved

Reply:
the middleware returns a promise and wrapping the whole code in a promise avoid the exceptions to "escape" the context and not get caught